### PR TITLE
python3Packages.morecantile: 5.3.0 -> 5.4.2

### PR DESCRIPTION
**Pull request by autra**
## Description of changes

python3Packages.morecantile: 5.3.0 -> 5.4.2

## Things done

- Built on platform(s)
  - [x] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [x] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://github.com/NixOS/nixpkgs/blob/master/pkgs/README.md#linking-nixos-module-tests-to-a-package) to the relevant packages
- [x] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2411.section.md) (or backporting [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) and [24.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

**Comment by autra** [comment]
Result of `nixpkgs-review pr 343354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

**Comment by sikmir** [comment]
On darwin:
```sh
error: builder for '/nix/store/3y0hvrzhbx1d6y7yrzafds1zw0d02iib-python3.12-morecantile-5.4.2.drv' failed with exit code 1;
       last 10 log lines:
       > >       assert tms.bbox == tms_bbox
       > E       assert BoundingBox(l...6368, top=0.0) == (100.23646734...50376368, 0.0)
       > E
       > E         At index 0 diff: 100.23646734667147 != 100.23646734667152
       > E         Use -v to get more diff
       >
       > tests/test_morecantile.py:430: AssertionError
       > =========================== short test summary info ============================
       > FAILED tests/test_morecantile.py::test_tiles_when_tms_bounds_and_provided_bounds_cross_antimeridian[bounds0-6-32750-tms_bbox0] - assert BoundingBox(l...6368, top=0.0) == (100.23646734...50376368, 0.0)
       > =================== 1 failed, 226 passed, 3 xfailed in 4.45s ===================
```

**Comment by autra** [comment]
Mmhh this looks like an upstream bug, what do you think?

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 343354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>4 packages failed to build:</summary>
  <ul>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>
<details>
  <summary>4 packages built:</summary>
  <ul>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
  </ul>
</details>

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 343354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

**Comment by sikmir** [comment]
> Mmhh this looks like an upstream bug, what do you think?

I think we can skip `test_tiles_when_tms_bounds_and_provided_bounds_cross_antimeridian` on darwin now and make a report to upstream.

**Comment by autra** [comment]
Result of `nixpkgs-review pr 343354` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

**Comment by autra** [comment]
@sikmir I've added a commit excluding this test for darwin. Can you check on your end? I don't have access to a mac...

File: pkgs/development/python-modules/morecantile/default.nix [hunk 1]
```diff
@@ -42,6 +43,15 @@ buildPythonPackage rec {
...
   ];
 
+  disabledTests =
+    if stdenv.isDarwin then
+      [
+        # touches network
+        "test_tiles_when_tms_bounds_and_provided_bounds_cross_antimeridian"
+      ]
+    else
+      [ ];
```
Comment (sikmir) [inline pkgs/development/python-modules/morecantile/default.nix:53]:
```suggestion
  disabledTests = lib.optionals stdenv.isDarwin [
    # https://github.com/developmentseed/morecantile/issues/156
    "test_tiles_when_tms_bounds_and_provided_bounds_cross_antimeridian"
  ];
```

**Comment by sikmir** [comment]
Result of `nixpkgs-review pr 343354` run on x86_64-darwin [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>8 packages built:</summary>
  <ul>
    <li>python311Packages.morecantile</li>
    <li>python311Packages.morecantile.dist</li>
    <li>python311Packages.rio-tiler</li>
    <li>python311Packages.rio-tiler.dist</li>
    <li>python312Packages.morecantile</li>
    <li>python312Packages.morecantile.dist</li>
    <li>python312Packages.rio-tiler</li>
    <li>python312Packages.rio-tiler.dist</li>
  </ul>
</details>

File: pkgs/development/python-modules/morecantile/default.nix (hunk 1 above)
Comment (autra) [reply pkgs/development/python-modules/morecantile/default.nix]: TIL `lib.optionals`, handy :-)

**Review by sikmir — APPROVED** [review]

**Pull request merged**